controllers/hostingdeploymentautoscalingjob/hostingdeploymentautoscalingjob_controller.go

```diff
@@ -102,12 +102,10 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return controllers.RequeueImmediately()
 	}
 
-	if err := r.reconcileHostingDeploymentAutoscalingJob(ctx); err != nil {
+	if err := r.reconcileHostingDeploymentAutoscalingJob(ctx); err != nil && !clientwrapper.IsHAP500Error(err) {
 		ctx.Log.Info("Got error while reconciling, will retry", "err", err)
-		//return controllers.RequeueImmediately()
-		return controllers.NoRequeue()
+		return controllers.RequeueImmediately()
```

Comment:
Why are we re-queuing if there isn't a 500 error? If there was a validation error input (a 400), this could get stuck in an infinite loop, and since these are immediate requeues it would be even worse? Should we requeue after interval instead?

Reply:
This was meant to be only for 5xx errors. so the not is in error. But the unit tests are checking this. Something is not right - thanks for the catch !

Reply:
for 5xx immediate requeues should be ok.